### PR TITLE
rootfs: start tee-supplicant on boot

### PR DESCRIPTION
**Pull request by jforissier**
Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>
Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (QEMU)

### CODE REVIEW

File: init.d.optee [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# Start/stop tee-supplicant (OP-TEE normal world daemon)
+#
+case "$1" in
+    start)
+	if [ -e /bin/tee-supplicant -a -e /dev/teepriv0 ]; then
+		echo "Starting tee-supplicant..."
+		tee-supplicant&
+		exit 0
+	else
+		echo "tee-supplicant or TEE device not found"
```
Comment (igoropaniuk) [inline init.d.optee]: minor: exit 1 (or another appropriate exit code)

File: init.d.optee [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+#
+# /etc/init.d/optee
+#
+# Start/stop tee-supplicant (OP-TEE normal world daemon)
+#
+case "$1" in
+    start)
+	if [ -e /bin/tee-supplicant -a -e /dev/teepriv0 ]; then
```
Comment (igoropaniuk) [inline init.d.optee]: Based on (https://github.com/OP-TEE/optee_client/blob/master/tee-supplicant/src/tee_supplicant.c#L388):
```
if (argc == 2) {
		fd = open_dev(argv[1]);
		if (fd < 0) {
			EMSG("failed to open \"%s\"", argv[1]);
			exit(EXIT_FAILURE);
		}
```
we can support different device file names. Maybe it's better to check in the ps list?

Comment (jbech-linaro) [inline init.d.optee]: 99.9% of the times when we're using tee-supplicant in our setup's we simply use `tee-supplicant &`. So, at this moment I don't think it's worth taking care of extra args etc, even though Igor is right in principle.

**Comment by jbech-linaro** [comment]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: init.d.optee (hunk 2 above)
Comment (jforissier) [inline init.d.optee]: Indeed, I'd rather keep this simple. Here we are targeting a controlled environment (our busybox-based rootfs) and we know that the TEE should be listening on `/dev/teepriv0`.

File: init.d.optee (hunk 1 above)
Comment (jforissier) [inline init.d.optee]: OK

File: init.d.optee (hunk 2 above)
Comment (igoropaniuk) [inline init.d.optee]: ok

**Comment by jforissier** [comment]
Updated.

**Comment by igoropaniuk** [comment]
`Reviewed-by: Igor Opaniuk <igor.opaniuk@linaro.org>`

**Pull request merged**